app/src/main/java/com/example/sharewhatyoucanproject/HomeFragment.kt

```diff
@@ -0,0 +1,133 @@
+        savedInstanceState: Bundle?
+    ): View? {
+        val view = inflater.inflate(R.layout.fragment_home, container, false)
+        uploadlayout = view.findViewById(R.id.uploadlayout)
+        val nameofuser = view.findViewById<TextView>(R.id.nameofuser)
+        nameofuser.text = "" + FirebaseAuth.getInstance().currentUser!!.displayName
+        checklocation = view.findViewById(R.id.checklocation)
+
+        convertbtn = view.findViewById(R.id.convertbtn)
+        locationRequest = com.google.android.gms.location.LocationRequest.create()
+        checklocation.setOnClickListener(View.OnClickListener {
+
+            if (context?.let { it1 ->
+                    ActivityCompat.checkSelfPermission(
+                        it1,
+                        Manifest.permission.ACCESS_FINE_LOCATION
+                    )
+                } != PackageManager.PERMISSION_GRANTED && context?.let { it1 ->
+                    ActivityCompat.checkSelfPermission(
+                        it1,
+                        Manifest.permission.ACCESS_COARSE_LOCATION
+                    )
+                } != PackageManager.PERMISSION_GRANTED
+            ) {
+
+
+            }
+            activity?.let { it1 ->
+                LocationServices.getFusedLocationProviderClient(it1)
+                    .requestLocationUpdates(locationRequest, object : LocationCallback() {
+                        override fun onLocationResult(locationresult: LocationResult) {
+                            super.onLocationResult(locationresult)
+
+                            if (locationresult.locations != null) {
+                                if (locationresult.getLocations().size > 0) {
+
+                                    val index: Int = locationresult.getLocations().size - 1
+
+                                    latitude =
+                                        locationresult.getLocations().get(index).getLatitude()
+                                    longitute =
+                                        locationresult.getLocations().get(index).getLongitude()
+
+                                    Toast.makeText(
+                                        context,
+                                        "latitude:- " + latitude + " " + "longitute: " + longitute,
+                                        Toast.LENGTH_SHORT
+                                    ).show()
+
+
+                                } else {
+                                    Toast.makeText(
+                                        context,
+                                        "size is 0: " + locationresult.locations.size,
+                                        Toast.LENGTH_SHORT
+                                    ).show()
+
+                                }
+                            } else {
+                                Toast.makeText(context, "it is null", Toast.LENGTH_SHORT).show()
+
+                            }
+
+                        }
```

Comment:
Should this, or part of this, or a superset of this, go in a helper function?  A little hard to read with all the indents and nested "ifs" and it being embedded in some other thing.  I'm having a hard time following this.